Mathlib/RingTheory/Finiteness.lean

```diff
@@ -776,20 +778,11 @@ theorem of_surjective (f : A →+* B) (hf : Surjective f) : f.Finite :=
   Module.Finite.of_surjective (Algebra.linearMap A B) hf
 
 theorem comp {g : B →+* C} {f : A →+* B} (hg : g.Finite) (hf : f.Finite) : (g.comp f).Finite := by
-  letI := f.toAlgebra
-  letI := g.toAlgebra
-  letI := (g.comp f).toAlgebra
-  letI : IsScalarTower A B C := RestrictScalars.isScalarTower A B C
-  letI : Module.Finite A B := hf
-  letI : Module.Finite B C := hg
+  algebraize f g (g.comp f)
   exact Module.Finite.trans B C
 
 theorem of_comp_finite {f : A →+* B} {g : B →+* C} (h : (g.comp f).Finite) : g.Finite := by
-  letI := f.toAlgebra
-  letI := g.toAlgebra
-  letI := (g.comp f).toAlgebra
-  letI : IsScalarTower A B C := RestrictScalars.isScalarTower A B C
-  letI : Module.Finite A C := h
+  algebraize f g (g.comp f)
```

Comment:
Why do we need to write `g.comp f` separately? Surely the tactic could identify composable pairs of specified RingHoms automatically?

Reply:
This would be nice to implement, but @callesonne I think it is also fine if you add this as

```
## Future work

* If source/target coincide, automatically add `(g.comp f)` when calling `algebraize f g`
```

to the module docstring if you don't feel like looking at how to implement it now.

Reply:
@semorrison we discussed this during the workshop, but an issue that comes up is that say you have three morphisms `f : A -> B` and `g, h : B -> C`. Then, if adding the algebra instance corresponding to the compositions is automatic, you would add two algebra instances `Algebra A C` which becomes a problem. So that's why we decided to force the user to supply the compositions themselves.